psyneulink/core/components/functions/transferfunctions.py

```diff
@@ -1482,13 +1514,23 @@ def _gen_llvm_transfer(self, builder, index, ctx, vi, vo, params, state):
         # Maxnum for some reason needs full function prototype
         max_f = ctx.get_builtin("maxnum", [ctx.float_ty])
         var = builder.load(ptri)
-        val = builder.fsub(var, bias)
-        val1 = builder.fmul(val, gain)
-        val2 = builder.fmul(val1, leak)
 
-        val = builder.call(max_f, [val1, val2])
+        if "derivative" in tags:
+            predicate = builder.fcmp_ordered('>', var, var.type(0))
+            with builder.if_else(predicate) as (then, otherwise):
+                with then:
+                    builder.store(gain, ptro)
+                with otherwise:
+                    val = builder.fmul(gain, leak)
+                    builder.store(val, ptro)
```

Comment:
`select` [0] would be more readable here

[0] https://llvmlite.readthedocs.io/en/latest/user-guide/ir/ir-builder.html#llvmlite.ir.IRBuilder.select